fedimint-api/src/tiered_multi.rs

```diff
@@ -141,19 +142,50 @@ where
 }
 
 impl TieredMulti<()> {
-    // TODO: move somewhere else?
-    pub fn represent_amount<K>(mut amount: Amount, tiers: &Tiered<K>) -> TieredMulti<()> {
-        let coins = tiers
-            .tiers()
-            .rev()
-            .map(|&amount_tier| {
-                let res = amount / amount_tier;
-                amount %= amount_tier;
-                (amount_tier, vec![(); res as usize])
-            })
-            .collect();
+    /// Determines the denominations to use when representing an amount
+    ///
+    /// Algorithm tries to leave the user with a target number of `denomination_sets` starting
+    /// at the lowest denomination.  `self` gives the denominations that the user already has.
+    pub fn represent_amount<K, V>(
+        amount: Amount,
+        current_denominations: &TieredMulti<V>,
+        tiers: &Tiered<K>,
+        denomination_sets: u16,
+    ) -> Tiered<usize> {
+        let mut remaining_amount = amount;
+        let mut denominations: HashMap<Amount, usize> = HashMap::new();
```

Comment:
Could we already make this a `Tiered` struct? `Tiered` seems to support `get_mut_or_default`, so should be possible? That way you wouldn't need to do the `denominations.into_iter().collect()` step in the end?

Reply:
I would still need to filter out the 0 entries, but I can change the type.